elastio-aws-resources-tagging/tag-elastio-resources.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+#usage info
+usage(){
+  echo "Usage:"
+  echo "  $0 -t tag"
+  echo
+  echo "Example:"
+  echo "  $0 -t tag1=value1 -t tag2=value2"
+  echo
+  exit
+}
+
+#parse options and arguments
+while getopts "t:" option; do
+  case $option in
+    "t")
+      arguments+=($(echo '{"Key":"'${OPTARG%=*}'","Value":"'${OPTARG#*=}'"}'))
+      ;;
+    *)
+      usage
+      ;;
+  esac
+done
+
+#no options were provided or there was a mistake
+if [ "$OPTIND" -eq "1" ] || [ "$OPTIND" -le "$#" ]; then
+  usage
+fi
+
+
+###add custom tags to ASGs
+
+customTagsASG=""
+
+for element in "${arguments[@]}"
+do
+
+ customTag=$(echo "Key=$(echo $element | jq ".Key" -r),Value=$(echo $element | jq ".Value" -r)")
+
+ #iterate through ASG with elastio in name
+ for ASGID in $(aws autoscaling describe-auto-scaling-groups --query "AutoScalingGroups[*].AutoScalingGroupName" --output text) ; \
+ do
+
+  if [[ "$ASGID" == *"elastio"* ]]; then
+
+   customTagsASG+=$(echo "ResourceId=$ASGID,ResourceType=auto-scaling-group,$customTag,PropagateAtLaunch=True ")
```

Comment:
This is an odd formulation.  Are you sure that `create-or-update-tags` works this way?  I'm on a plane w/ shitty internet right now so I can't check it myself but it looks suspicious.

Reply:
Yes. I was surprised as well. Here is how the command look:
```
aws autoscaling create-or-update-tags \
    --tags ResourceId=my-asg,ResourceType=auto-scaling-group,Key=Role,Value=WebServer,PropagateAtLaunch=true ResourceId=my-asg,ResourceType=auto-scaling-group,Key=Dept,Value=Research,PropagateAtLaunch=true
```
Autoscaling group names and tags are specified in the `--tag`.